### ISSUE TITLE
Any way to get total posts count?

### ISSUE THREAD

**Issue by proleskovskiy** [wp-net/WordPressPCL#255]
Hello!

According to the official docs:
https://developer.wordpress.org/rest-api/using-the-rest-api/pagination/

"To determine how many pages of data are available, the API returns two header fields with every paginated response:
    X-WP-Total: the total number of records in the collection
    X-WP-TotalPages: the total number of pages encompassing all available records"

I query posts using `await client.Posts.Query(query)`, with 'Page' and 'PerPage' parameters set, but this method only returns posts. Is there a way to get total posts count without actually loading all of them with `client.Posts.GetAll()`?

**Comment by ThomasPe** [comment]
I agree that this would be a useful addition, I'll add it to the backlog.

**Comment by robertsundstrom** [comment]
Perhaps, instead of returning and IEnumerable<T> for Queries, instead return a custom collection with additional properties related to the response, including the page count.

**Comment by robertsundstrom** [comment]
My prototype: https://github.com/robertsundstrom/WordPressPCL/tree/queryresult

**Comment by robertsundstrom** [comment]
Pull request: https://github.com/wp-net/WordPressPCL/pull/257

**Comment by robertsundstrom** [comment]
I have published a sample app here:

https://github.com/robertsundstrom/headless-wordpress-blazor

The Frontend project contains a build of WordPressPCL.

**Comment by ThomasPe** [comment]
Hey, thanks so much for your work. I'll try to check it out on the weekend!

**Comment by sanishmistry** [comment]
I also have this issue.

**Comment by delasource** [comment]
please check this. i'm looking for an easy way of pagination to get all items (not just Posts but custom object types)

An easy way would be just to expose HttpClient.LastResponseHeaders publicly. (of course this may not be the best thread-safe solution)

**Comment by ThomasPe** [comment]
Closing as resolved or superseded by the current implementation/documentation. Please reopen with current-version details if this is still needed.
